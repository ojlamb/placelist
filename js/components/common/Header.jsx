import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import PropTypes from "prop-types";
import compose from "recompose/compose";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import AccountCircle from "@material-ui/icons/AccountCircle";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";
import * as sessionActions from "../../actions/sessionActions";

const styles = {
  root: {
    flexGrow: 1
  },
  flex: {
    flexGrow: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  button: {
    color: "white"
  },
  link: {
    textDecoration: "none",
    color: "white"
  }
};

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null
    };
    this.logOut = this.logOut.bind(this);
  }

  logOut(event) {
    event.preventDefault();
    this.props.actions.logOutUser();
    this.setState({ anchorEl: null });
  }

  handleMenu = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { classes } = this.props;
    const open = Boolean(this.state.anchorEl);

    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <Typography
              variant="title"
              color="inherit"
              className={classes.flex}
            >
              <Link className={classes.link} to="/">
                PlaceList
              </Link>
            </Typography>
            {this.props.session ? (
              <div>
                <Link className={classes.link} to="/map">
                  <Button className={classes.button}>Map</Button>
                </Link>
                <Link className={classes.link} to="/places">
                  <Button className={classes.button} to="places">
                    List
                  </Button>
                </Link>
                <Link className={classes.link} to="/about">
                  <Button className={classes.button}>About</Button>
                </Link>
                <IconButton
                  aria-owns={open ? "menu-appbar" : null}
                  aria-haspopup="true"
                  onClick={this.handleMenu}
                  color="inherit"
                >
                  <AccountCircle />
                </IconButton>
                <Menu
                  id="menu-appbar"
                  anchorEl={this.state.anchorEl}
                  anchorOrigin={{
                    vertical: "top",
                    horizontal: "right"
                  }}
                  transformOrigin={{
                    vertical: "top",
                    horizontal: "right"
                  }}
                  open={open}
                  onClose={this.handleClose}
                >
                  <MenuItem onClick={this.handleClose}>Profile</MenuItem>
                  <MenuItem onClick={this.handleClose}>My account</MenuItem>
                  <MenuItem onClick={this.logOut}>Log Out</MenuItem>
                </Menu>
              </div>
            ) : (
              <div>
                <div>
                  <Link className={classes.link} to="/login">
                    <Button
                      variant="outlined"
                      color="primary"
                      className={classes.button}
                    >
                      Login
                    </Button>
                  </Link>
                  <Link className={classes.link} to="/signup">
                    <Button className={classes.button}>Sign Up</Button>
                  </Link>
                </div>
              </div>
            )}
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  session: state.session
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(sessionActions, dispatch)
});

Header.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  actions: PropTypes.shape({
    logOutUser: PropTypes.func.isRequired
  }).isRequired,
  session: PropTypes.bool.isRequired
};

export default compose(
  withStyles(styles),
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(Header);
