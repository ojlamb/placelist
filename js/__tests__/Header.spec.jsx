import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Header from "../components/common/Header";

test("Header renders correctly", () => {
  const component = renderer.create(<Header />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
