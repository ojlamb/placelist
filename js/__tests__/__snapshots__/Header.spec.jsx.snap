// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-6 MuiAppBar-positionStatic-10 MuiAppBar-colorPrimary-12"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <h2
        className="MuiTypography-root-45 MuiTypography-title-51 MuiTypography-colorInherit-64 Header-flex-2"
      >
        <button
          className="MuiButtonBase-root-96 MuiButton-root-70 MuiButton-text-72 MuiButton-flat-75 Header-link-5"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          to="/"
          type="button"
        >
          <span
            className="MuiButton-label-71"
          >
            PlaceList
          </span>
          <span
            className="MuiTouchRipple-root-99"
          />
        </button>
      </h2>
      <div>
        <button
          className="MuiButtonBase-root-96 MuiButton-root-70 MuiButton-text-72 MuiButton-flat-75 Header-button-4"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          to="map"
          type="button"
        >
          <span
            className="MuiButton-label-71"
          >
            Map
          </span>
          <span
            className="MuiTouchRipple-root-99"
          />
        </button>
        <button
          className="MuiButtonBase-root-96 MuiButton-root-70 MuiButton-text-72 MuiButton-flat-75 Header-button-4"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          to="list"
          type="button"
        >
          <span
            className="MuiButton-label-71"
          >
            List
          </span>
          <span
            className="MuiTouchRipple-root-99"
          />
        </button>
        <button
          className="MuiButtonBase-root-96 MuiButton-root-70 MuiButton-text-72 MuiButton-flat-75 Header-button-4"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          to="about"
          type="button"
        >
          <span
            className="MuiButton-label-71"
          >
            About
          </span>
          <span
            className="MuiTouchRipple-root-99"
          />
        </button>
        <button
          aria-haspopup="true"
          aria-owns={null}
          className="MuiButtonBase-root-96 MuiIconButton-root-106 MuiIconButton-colorInherit-107"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-111"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-112"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-99"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
